BL-125 Feature/be As a user I can enter a search query to search asylum case decision data

EricMarshallLambert opened the pull request:
## Description

We configured the API controller to allow users to send query parameters as GET requests to the /cases endpoint. The controller was updated with a request mapped method that accepts CasesRequestDto as an argument. Tests were stubbed out, but waiting on the service implementation before completion. Project was updated to exclude raceOrEthnicity fields. 

#### Video Link

https://www.loom.com/share/468aa8b4e1e347e8acc40875a06154d0

This task required research to understand if the controller could map requests to the Data Transfer Object (DTO) that was created. The Spring Boot controller layer will map the query parameter values from the request at runtime when a request is received. This is done by providing the method a RequestDTO as an argument. There is no need to explicitly denote the query parameters in the method signature as we have done in previous projects. The /cases request is automatically converted into the CasesRequestDto at runtime . The RequestDto then can be passed into the service layer for processing.
Note: The fields within the Request DTO must not be of primitive type for automatic conversion. The conversion from a string leverages the wrapper class methods.

#### Jira Link
https://bloomtechlabs.atlassian.net/browse/BL-125
<blockquote class="jira-card"><a href="https://bloomtechlabs.atlassian.net/browse/BL-125">BL-125 Feature/be As a user I can enter a search query to search asylum case decision data</a></blockquote>

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

willkim76 committed:
Deprecated All RELEASE_1 Cases Endpoints. Defined a new Endpoint for Query Search. Defined a new DTO for Query Parameters.

willkim76 committed:
Updated the AsylumCaseControllerTest for new Endpoint.

willkim76 committed:
Updated the AsylumCaseControllerTest for new Endpoint. Deprecated legacy test cases.

willkim76 committed:
Merge branch 'BL-125-be-as-a-user-i-can-enter-a-search-query-to-search-asylum-case-decision-data' of https://github.com/BloomTech-Labs/asylum-rg-be-a into BL-125-be-as-a-user-i-can-enter-a-search-query-to-search-asylum-case-decision-data

willkim76 committed:
Updated QueryParametersDto to CasesRequestDto. Updated AsylumCase Entity: Removed Race/Ethnicity Attribute. Updated Seeder.

attaris978 commented on src/test/java/com/bloomtech/asylumrgbea/controllers/AsylumCaseControllerTest.java:
```diff
@@ -42,23 +42,47 @@ void getAllCases_serviceIsValid_returnsIterableOfAsylumCaseResponseDto() {
+        // TODO: AsylumCaseServiceImpl has not been scoped-redesigned.
+    }
+
+    void getCases_withSingleQueryParameterNoMatches_returnsErrorDto() {
```
This division of potential outcomes will prove helpful in fleshing out frontend handling of ErrorDTOs. A consideration for implementation, when that happens, should be whether the query parameters provided are viable values -- we have discreet values for most of the potential inputs, and so validating these values on the backend may prove invaluable in detecting aberrations between front end and back end schema.

attaris978 reviewed (dismissed):
Excellent forethought and intuitive structuring of code. Be sure to confer with the front end when designing ErrorDTOs so that the frontend will know what to expect for a response and can process it accordingly.
More importantly, this feels like progress. Love it!

shannonsullivan reviewed (dismissed):
PR looks good. The description is well documented along with the video explaining what has been done and why. It was helpful to see an old project example compared with this project demonstrating the advantages of using Spring Boot. Great forward thinking on stubbing out unit tests.

vargas-santiagoo reviewed (changes requested):
Everything looks good but now that the [`AsylumCase.java`](https://github.com/BloomTech-Labs/asylum-rg-be-a/blob/BL-125-be-as-a-user-i-can-enter-a-search-query-to-search-asylum-case-decision-data/src/main/java/com/bloomtech/asylumrgbea/entities/AsylumCase.java) entity does not have a `raceOrEthnicity` value the [`AsylumCaseResponseDto.java`](https://github.com/BloomTech-Labs/asylum-rg-be-a/blob/BL-125-be-as-a-user-i-can-enter-a-search-query-to-search-asylum-case-decision-data/src/main/java/com/bloomtech/asylumrgbea/models/AsylumCaseResponseDto.java) should reflect that change.

EricMarshallLambert committed:
BL-125
Update AsylumCaseResponseDto.java to reflect changes decided to not use raceOrEthnicity

BeatriceMani reviewed (dismissed):
Really clear walkthrough on the Loom nice to include the comparison to our previous use of Spring. This looks good, and sets up our future work well!

ashtilawat commented on src/main/java/com/bloomtech/asylumrgbea/controllers/AsylumCaseController.java:
```diff
@@ -33,11 +41,31 @@ public PageResponseDto getPageOfCases(@RequestBody AsylumCaseRequestDto asylumCa
+     * to obscure this endpoint from public access.
      */
-    @GetMapping
+    @Deprecated
```
Let's remove deprecated endpoints from the codebase.

EricMarshallLambert replied:
Removed the endpoints.

ashtilawat reviewed (changes requested):
Let's remove deprecated endpoints.

EricMarshallLambert committed:
BL-125
Update removed all methods that were previously deprecated including the all tests.

dkav1122 reviewed (approved):
Solid refactoring of legacy code to remove unnecessary code/methods that don't make sense for our current design. This will make working with our codebase better for the future, and set the precedent for maintaining a clean code base.

shannonsullivan reviewed (approved):
PR looks good. As discussed, all deprecated endpoints were removed to simplify code.

enpk reviewed (approved):
Code looks good, well setup to move forward with our new endpoint.

BeatriceMani reviewed (approved):
Motivation for the changes was clearly laid out. Removal of deprecated endpoint simplifies the code.
